torch/_inductor/codegen/triton.py

```diff
@@ -2054,6 +2054,22 @@ def load(self, name: str, index: sympy.Expr):
         #   3.2) Its not its last use
         #   3.3) This load will not be lifted to the body
         #
+        """Check if the buffer we're about to load, has
+        more than one read dependency
+        """
+        depcount = 0
+        has_read_deps = (
+            True  # by default, assume that this buffer has > 1 read dependencies
+        )
+        if (
+            config.triton.skip_l1_cache and self.current_node
+        ):  # it can be that self.current_node is None (!)
+            for node in self.current_node.scheduler.nodes:
+                for l in list(node.read_writes.reads):
+                    if l.name == name:
+                        depcount += 1
+            has_read_deps = depcount > 1  # has more than one read dep
+
```

Comment:
This is O(n^2), because we loop over all reads once every load.  The result is also only used if many conditions are met, so this will be unused most of the time.

We could make this O(n) computing a dict:
```
buffer_read_counts = {buffer_name: count, ....} 
```

I'd suggest putting this in simd_kernel_features.py and accessing with `self.features.buffer_read_counts()` use `@cache_on_self` so it is computed once lazily (so kernels that don't need it don't compute it).